src/domains/meshes/cartesiangrid.jl

```diff
@@ -50,23 +50,16 @@ Create a 1D grid from -1 to 1 with 100 segments:
 julia> CartesianGrid((-1.0,), (1.0,), dims=(100,))
 ```
 """
-struct CartesianGrid{C<:CRS,Mₚ<:Manifold,Dim,ℒ<:Len} <: Grid{𝔼{Dim},C,Dim}
-  origin::Point{Mₚ,C}
-  spacing::NTuple{Dim,ℒ}
-  offset::Dims{Dim}
-  topology::GridTopology{Dim}
+const CartesianGrid{M<:𝔼,C<:Cartesian} = RegularGrid{M,C}
 
-  function CartesianGrid(
-    origin::Point{Mₚ,C},
-    spacing::NTuple{Dim,ℒ},
-    offset::Dims{Dim},
-    topology::GridTopology{Dim}
-  ) where {C<:CRS,Mₚ<:Manifold,Dim,ℒ<:Len}
-    if !all(>(zero(ℒ)), spacing)
-      throw(ArgumentError("spacing must be positive"))
-    end
-    new{C,Mₚ,Dim,float(ℒ)}(origin, spacing, offset, topology)
-  end
+function CartesianGrid(
+  origin::Point{M,C},
+  spacing::NTuple{Dim,ℒ},
+  offset::Dims{Dim},
+  topology::GridTopology{Dim}
+) where {M<:𝔼,C<:Cartesian,Dim,ℒ<:Len}
+  sp = float.(spacing)
```

Comment:
this conversion to float should happen in the RegularGrid constructor, it doesn't seem specific to CartesianGrid